Update Toolkit to work with Trados Studio 2022 and Trados Studio 2024…

aflorescu579774 opened the pull request:
Changes Made

Add DependencyLoader Module to help with the conflicts between Powershell dependencies version and Trados dependencies version
Removed the Get-DefaultProjectTemplate and Create a project object just with the ProjectInfo parameter, Factory.CreateApplication() not returning the properties that were used in the previous versions 
Validate-TaskSequence uses now a [SDL.ProjectAutomation.Core.TaskSequence] object as this is returned now by the Project.RunAutomaticTask 
Overall update the toolkit to work with Trados Studio 2022 and Trados Studio 2024 
Added command prompt scripts for 2022 and 2024

aflorescu579774 committed:
Update Toolkit to work with Trados Studio 2022 and Trados Studio 2024 Changes Made
-> Add DependencyLoader Module to help with the conflicts between Powershell dependencies version and Trados dependencies version
-> Removed the Get-DefaultProjectTemplate and Create a project object just with the ProjectInfo parameter, Factory.CreateApplication() not returning the properties that were used in the previous versions
-> Validate-TaskSequence uses now a [SDL.ProjectAutomation.Core.TaskSequence] object as this is returned now by the Project.RunAutomaticTask
-> Overall update the toolkit to work with Trados Studio 2022 and Trados Studio 2024
-> Added command prompt scripts for 2022 and 2024

EvzenP commented:
Good to see that STraSAK is still alive after all those years.  
However, I don't get the removal of `Get-DefaultProjectTemplate` - does it mean that Studio has fixed that long-standing bug with default project template in API (as described in the removed code)? Or you have simply removed the "code which does not work anymore" without replacing its functionality?

aflorescu579774 committed:
SDLCOM-5928 - Updated the Toolkit for Project Creation, The project creation by default is using the default project template set in Trados Studio

aflorescu579774 committed:
Update ProjectHelper.psm1

Change the source and target language parameters from mandatory to not mandatory
